js/components/password.js

```diff
@@ -43,27 +43,65 @@ Fliplet.FormBuilder.field('password', {
     },
     description: {
       type: String
+    },
+    isPasswordsSame: {
+      type: Boolean,
+      default: true
+    },
+    isValid: {
+      type: Boolean,
+      default: true
     }
   },
+  data: function() {
+    return {
+      isFocused: false,
+      isPreview: Fliplet.Env.get('preview'),
+      passwordMinLength: 8,
+      defaultClass: 'panel-default',
+      invalidClass: 'panel-danger',
+      rules: {
+        isUppercase: new RegExp('[A-Z]'),
+        isLowercase: new RegExp('[a-z]'),
+        isNumber: new RegExp('[0-9]'),
+        isSpecial: new RegExp('[^A-Za-z0-9]')
+      }
+    };
+  },
   validations: function() {
     var rules = {
-      value: {},
-      passwordConfirmation: {}
+      value: {
+        required: this.required && window.validators.required,
+        minLength: window.validators.minLength(this.passwordMinLength),
+        containsUppercase: function(value) {
+          return value ? this.rules.isUppercase.test(value) : true;
```

Comment:
If there's no value, why is the fallback `true`? Surely if there's no value, then it doesn't contain anything uppercase, and therefore the fallback should be `false`?